domain/externalcontroller/service/service.go

```diff
@@ -7,25 +7,40 @@ import (
 	"context"
 
 	"github.com/juju/errors"
+	"github.com/juju/loggo"
 
+	"github.com/juju/juju/core/changestream"
 	"github.com/juju/juju/core/crossmodel"
+	"github.com/juju/juju/core/watcher/eventsource"
+	"github.com/juju/juju/domain"
 )
 
+var logger = loggo.GetLogger("juju.domain.service.externalcontroller")
```

Comment:
We shouldn't pass in loggers like this. Instead, they should be passed in via the constructor.